Source/LedDriver.c

```
@@ -0,0 +1,97 @@
+#include "LedDriver.h"
+#include "uassert.h"
+
+static uint16_t * ledAddress;
+static uint16_t ledImage;
+
+enum {ALL_LEDS_ON = ~0, ALL_LEDS_OFF = ~ALL_LEDS_ON};
+enum {FIRST_LED = 1, LAST_LED = 16 };
+
+static BOOL IsLedOutOfBounds(int ledNumber)
+{
+    if ((ledNumber < FIRST_LED) || (ledNumber > LAST_LED))
+    {
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+static uint16_t convertLedNumberToBit(int ledNumber)
+{
+    return 1 << (ledNumber - 1);
+}
+
+static void updateHardware(void) 
+{
+    *ledAddress = ledImage;
+}
+
+void LedDriver_Create(uint16_t * address) 
+{
+    ledAddress = address;
+    ledImage = ALL_LEDS_OFF;
+    updateHardware();
+}
+
+void LedDriver_Destroy(void)
+{
+}
+
+static void setLedImageBit(int ledNumber)
+{
+    ledImage |= convertLedNumberToBit(ledNumber);
+}
+void LedDriver_TurnOn(int ledNumber)
+{
+    if(IsLedOutOfBounds(ledNumber))
```

so looks like we are safe due to your check!

So I have one comment on this.  Making this call with invalid data actually returns and does nothing.  Typically what we would do instead is:

```uassert(LedIsValid()```

This is because if there is an issue you want to fail fast so you can fix the problem.